### PR TITLE
daemon: Fix endpoint restore log regarding health endpoint

### PR DESCRIPTION
**Pull request by manalibhutiyani**
On each restart, the health endpoint of the previous run cannot be restored.
This is fine as it is supposed to be recreated but it triggers an unnecessary
info message about restore not succeeding.

The cleaning of the state was already taking place as part of
health endpoint being added to `state.toClean`.
We simply need to get rid of the unnecessary log printed.

Fixes: #4325
Fixes: 403a2ec agent: Reserve existing endpoint IPs before allocating auxiliary IPs
Signed-off-by: Manali Bhutiyani <manali@covalent.io>

```release-note
Fix endpoint restore log regarding health endpoint
```

---
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/4561)

### CODE REVIEW

**Comment by manalibhutiyani** [comment]
test-me-please

File: daemon/state.go [hunk 1]
```diff
@@ -77,11 +77,14 @@ func (d *Daemon) restoreOldEndpoints(dir string, clean bool) (*endpointRestoreSt
 	for _, ep := range possibleEPs {
 		scopedLog := log.WithField(logfields.EndpointID, ep.ID)
 		skipRestore := false
+
 		if _, err := netlink.LinkByName(ep.IfName); err != nil {
 			scopedLog.Infof("Interface %s could not be found for endpoint being restored, ignoring", ep.IfName)
 			skipRestore = true
 		} else if !workloads.IsRunning(ep) {
-			scopedLog.Info("No workload could be associated with endpoint being restored, ignoring")
+			if ep.SecurityIdentity.ID != identityPkg.ReservedIdentityHealth {
```
Comment (ianvernon) [inline daemon/state.go]: Small nit - add a comment specifying why we ignore the health endpoint to give context to those who will edit the code in the future.

Comment (manalibhutiyani) [reply daemon/state.go]: sure.

Comment (manalibhutiyani) [reply daemon/state.go]: done. Please take a look

**Comment by manalibhutiyani** [comment]
test-me-please

**Review by ianvernon — APPROVED** [review]

**Comment by manalibhutiyani** [comment]
test-me-please

**Review by joestringer — APPROVED** [review]

**Review by aanm — DISMISSED** [review]
@manalibhutiyani Can you add the commit ID this is fixing? We need to make sure if this affect 1.0 or not

**Comment by manalibhutiyani** [comment]
@aanm This log was added some time back as part of commit `403a2ec16de6de96b2fa68894eab7b7311761938` titled `agent: Reserve existing endpoint IPs before allocating auxiliary IPs`
https://github.com/cilium/cilium/commit/403a2ec16de6de96b2fa68894eab7b7311761938


Should I simply add 
`Fixes: 403a2ec16de6de96b2fa68894eab7b7311761938` in the commit
? Or something else.

**Comment by manalibhutiyani** [comment]
test-me-please

**Review by aanm — APPROVED** [review]

File: daemon/state.go [hunk 2]
```diff
@@ -77,11 +77,16 @@ func (d *Daemon) restoreOldEndpoints(dir string, clean bool) (*endpointRestoreSt
...
 		skipRestore := false
+
 		if _, err := netlink.LinkByName(ep.IfName); err != nil {
 			scopedLog.Infof("Interface %s could not be found for endpoint being restored, ignoring", ep.IfName)
 			skipRestore = true
 		} else if !workloads.IsRunning(ep) {
-			scopedLog.Info("No workload could be associated with endpoint being restored, ignoring")
+			// On each restart, the health endpoint is supposed to be recreated.
+			// Hence we do not need to explicitly log a message in this case.
+			if ep.SecurityIdentity.ID != identityPkg.ReservedIdentityHealth {
```
Comment (tgraf) [inline daemon/state.go]: Please move this check to before line 80. We shouldn't handle this in the !IsRunning case only but clean the health endpoint unconditionally.

**Review by tgraf — DISMISSED** [review]

File: daemon/state.go (hunk 2 above)
Comment (manalibhutiyani) [reply daemon/state.go]: @tgraf : 
We need to still execute the 
```
if clean && skipRestore {
			state.toClean = append(state.toClean, ep)
			continue
		}
```
So changing the logic to following should work:
```
	// On each restart, the health endpoint is supposed to be recreated.
		// Hence we need to clean health endpoint unconditionally
		if ep.SecurityIdentity.ID == identityPkg.ReservedIdentityHealth {
			skipRestore = true
		} else {
			if _, err := netlink.LinkByName(ep.IfName); err != nil {
				scopedLog.Infof("Interface %s could not be found for endpoint being restored, ignoring", ep.IfName)
				skipRestore = true
			} else if !workloads.IsRunning(ep) {
				scopedLog.Info("No workload could be associated with endpoint being restored, ignoring")
				skipRestore = true
			}
		}

		if clean && skipRestore {
			state.toClean = append(state.toClean, ep)
			continue
		}
```

Comment (tgraf) [reply daemon/state.go]: Please use `} else if {`

Comment (manalibhutiyani) [reply daemon/state.go]: Or simply clean it unconditionally ignoring `clean` bool:

```
for _, ep := range possibleEPs {
		scopedLog := log.WithField(logfields.EndpointID, ep.ID)
		skipRestore := false

		// On each restart, the health endpoint is supposed to be recreated.
		// Hence we need to clean health endpoint state unconditionally.
		if ep.SecurityIdentity.ID == identityPkg.ReservedIdentityHealth {
			state.toClean = append(state.toClean, ep)
			continue
		}
		if _, err := netlink.LinkByName(ep.IfName); err != nil {
			scopedLog.Infof("Interface %s could not be found for endpoint being restored, ignoring", ep.IfName)
			skipRestore = true
		} else if !workloads.IsRunning(ep) {
			scopedLog.Info("No workload could be associated with endpoint being restored, ignoring")
			skipRestore = true
		}

		if clean && skipRestore {
			state.toClean = append(state.toClean, ep)
			continue
		}
```

Comment (manalibhutiyani) [reply daemon/state.go]: Going with the first suggestion. Let me know if you think the second option i.e ignoring `clean` and unconditionally cleaning is a better option.

**Comment by manalibhutiyani** [comment]
test-me-please

File: daemon/state.go (hunk 2 above)
Comment (manalibhutiyani) [reply daemon/state.go]: done. Please check

**Comment by manalibhutiyani** [comment]
test-me-please

**Review by tgraf — APPROVED** [review]

**Pull request merged**